### PR TITLE
Update to XPRESS 9.0

### PR DESCRIPTION
**Pull request by pet-mit**
It is actually difficult to keep track of control parameters offered by different XPRESS versions and map them in the code.
- The python script has been updated to make it easier to update the control parameters when updating XPRESS version (added a mention in the readme)
- The code has been updated to map new control parameters of XPRESS 9.0
- Int64 controls are now tested sperately
- XPRS_COMPUTE control parameter has been disabled: the unit test shows it cannot be changed properly (the previous unit test before this PR tested setting it to 0, which is its default value). This seems to be a parameter for extreme power users, so it should not be an issue.
- the workflows have been updated to explicitly specify the XPRESS version used

### CODE REVIEW

File: ortools/xpress/parse_header_xpress.py [hunk 1]
```diff
@@ -145,12 +109,36 @@ def __init__(self):
...
+        if symbol in self.__excluded_defines:
+            print('skipping ' + symbol)
+            return
+
+        # If it is a control parameter, import it to expose it to the user
+        # Else import it only if required
+        if self.__doc_section == XprsDocumentSection.STRING_PARAMS \
+                or self.__doc_section == XprsDocumentSection.DOUBLE_PARAMS \
+                or self.__doc_section == XprsDocumentSection.INT_PARAMS \
+                or self.__doc_section == XprsDocumentSection.INT64_PARAMS:
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:126]:
```suggestion
        if self.__doc_section in [XprsDocumentSection.STRING_PARAMS, \
                                  XprsDocumentSection.DOUBLE_PARAMS
                                  # etc.
                                  ]
```

File: ortools/xpress/parse_header_xpress.py [hunk 2]
```diff
@@ -172,12 +160,34 @@ def parse(self, filepath):
...
             if not line:  # Ignore empty lines.
                 continue
-            if re.match(r'/\*', line, re.M):  # Ignore comments.
+
+            if " * control parameters for XPRSprob" in line:
+                XPRSprob_section = True
                 continue
 
+            if XPRSprob_section and "/***************************************************************************\\" in line:
+                XPRSprob_section = False
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:174]: A bit weird, care to explain ?

File: ortools/xpress/parse_header_xpress.py [hunk 3]
```diff
@@ -145,12 +109,36 @@ def __init__(self):
...
+                or self.__doc_section == XprsDocumentSection.INT_PARAMS \
+                or self.__doc_section == XprsDocumentSection.INT64_PARAMS:
+            self.__header += f'#define {symbol} {value}\n'
+            ortools_symbol = symbol.replace("XPRS_", "")
+            if self.__doc_section == XprsDocumentSection.STRING_PARAMS:
+                self.__string_parameters += f'{{\"{ortools_symbol}\", {symbol}}},\n'
+                self.__string_parameters_test += f'{{\"{ortools_symbol}\", {symbol}, "default_value"}},\n'
+            elif self.__doc_section == XprsDocumentSection.DOUBLE_PARAMS:
+                self.__double_parameters += f'{{\"{ortools_symbol}\", {symbol}}},\n'
+                self.__double_parameters_test += f'{{\"{ortools_symbol}\", {symbol}, 1.}},\n'
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:134]:
```suggestion
                self.__double_parameters_unittest += f'{{\"{ortools_symbol}\", {symbol}, 1.}},\n'
```

File: ortools/xpress/parse_header_xpress.py (hunk 2 above)
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:174]: Encapsulate in a function that returns a `bool`

Comment (pet-mit) [reply ortools/xpress/parse_header_xpress.py:174]: we only handle XPRESS object XPRSprob, so we only have to import its control parameters. the xprs.h header file contains control parameters for XPRSprob but also for XPRSmipsolpool, XPRSmipsolenum.. this variable detects the XPRSprob sections by parsing comments and detecting sections (delimited by "*****"..)

Comment (pet-mit) [reply ortools/xpress/parse_header_xpress.py]: I added a comment in the new method

**Review by flomnes — APPROVED** [review]

**Pull request merged**